examples/permission/asbtract-class.ts

```diff
@@ -0,0 +1,87 @@
+import { IPermissionProvider, IPermissionHandler, Permissions, PluginProfile, ModuleProfile } from '../../src/types'
+
+
+/**
+ * Example of a PermissionHandler using localStorage
+ */
+export abstract class SecurityHandler implements IPermissionHandler, IPermissionProvider {
+  public permissions: Permissions
+  abstract confirm(
+    message: string,
+  ): Promise<{ allow: boolean; remember: boolean }>
+
+  constructor() {
+    const permission = localStorage.getItem('permissions')
+    this.permissions = permission ? JSON.parse(permission) : {}
+  }
+
+  private setPermissions() {
```

Comment:
could we have a different naming here?
like `persistPermissions`

Reply:
yes, this is an example. You don't actually need to implement this method. I need it here to avoid having a getter/setter directly on `localStorage`. I'll change it in the example to make it clearer to understand though. Thanks.